Revisions not working: getting React error

### Link to reproduction

private repo

### Describe the Bug

Opening the revisions page produces a React error in any collection with revisions activated.

Example:

http://localhost:3000/admin/globals/site/versions/6568de9ae04b3fc4b44c739c

React error:

Objects are not valid as a React child (found: object with keys {de, en, fr, it}). If you meant to render a collection of children, use an array instead.

From the error message an object with translation data is added to the JSX tree. After that the SPA is broken i.e. back navigation fails. We can reproduce it on our development and production systems. We tried to get more details but the React developer tools show no info in this case.


### To Reproduce

Open revisions detail page. Getting blank screen with error in JavaScript console.

### Payload Version

2.3.1

### Adapters and Plugins

db-mongodb, bundler-webpack, live-preview

Full stack trace:

```
main.aa84253b4ab011b8cd39.js:63 Error: Minified React error #31; visit https://reactjs.org/docs/error-decoder.html?invariant=31&args[]=object%20with%20keys%20%7Bde%2C%20en%2C%20fr%2C%20it%7D for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at aJ (main.aa84253b4ab011b8cd39.js:63:54264)
    at l (main.aa84253b4ab011b8cd39.js:63:59257)
    at il (main.aa84253b4ab011b8cd39.js:63:71124)
    at u (main.aa84253b4ab011b8cd39.js:63:121237)
    at lR (main.aa84253b4ab011b8cd39.js:63:99083)
    at main.aa84253b4ab011b8cd39.js:63:98950
    at lz (main.aa84253b4ab011b8cd39.js:63:98957)
    at lO (main.aa84253b4ab011b8cd39.js:63:95709)
    at lS (main.aa84253b4ab011b8cd39.js:63:96098)
    at r3 (main.aa84253b4ab011b8cd39.js:63:44823)
```

I tried to find the component but it's hard to find any details in the minified code.

--- a/packages/payload/src/admin/components/views/Version/RenderFieldsToDiff/index.tsx
+++ b/packages/payload/src/admin/components/views/Version/RenderFieldsToDiff/index.tsx
@@ -43,11 +43,11 @@ const RenderFieldsToDiff: React.FC<Props> = ({
           if (field.localized) {
             return (
               <div className={`${baseClass}__field`} key={i}>
-                {locales.map((locale) => {
+                {locales.map((locale, index) => {
                   const versionLocaleValue = versionValue?.[locale]
                   const comparisonLocaleValue = comparisonValue?.[locale]
                   return (
-                    <div className={`${baseClass}__locale`} key={locale}>
+                    <div className={`${baseClass}__locale`} key={[locale, index].join('-')}>
                       <div className={`${baseClass}__locale-value`}>
                         <Component
                           comparison={comparisonLocaleValue}

--- a/packages/payload/src/admin/components/views/Version/SelectLocales/index.tsx
+++ b/packages/payload/src/admin/components/views/Version/SelectLocales/index.tsx
@@ -4,12 +4,26 @@ import { useTranslation } from 'react-i18next'
 import type { Props } from './types'
 
 import ReactSelect from '../../../elements/ReactSelect'
+import { useLocale } from '../../../utilities/Locale'
 import './index.scss'
 
 const baseClass = 'select-version-locales'
 
 const SelectLocales: React.FC<Props> = ({ onChange, options, value }) => {
   const { t } = useTranslation('version')
+  const { code } = useLocale()
+
+  const format = (items) => {
+    return items.map((item) => {
+      if (typeof item.label === 'string') return item
+      if (typeof item.label !== 'string' && item.label[code]) {
+        return {
+          label: item.label[code],
+          value: item.value,
+        }
+      }
+    })
+  }
 
   return (
     <div className={baseClass}>
@@ -17,9 +31,9 @@ const SelectLocales: React.FC<Props> = ({ onChange, options, value }) => {
       <ReactSelect
         isMulti
         onChange={onChange}
-        options={options}
+        options={format(options)}
         placeholder={t('selectLocales')}
-        value={value}
+        value={format(value)}
       />
     </div>
   )

